pkg/controlplane/reconcilers/lease.go

```diff
@@ -305,5 +308,6 @@ func (r *leaseEndpointReconciler) RemoveEndpoints(serviceName string, ip net.IP,
 func (r *leaseEndpointReconciler) StopReconciling() {
 	r.reconcilingLock.Lock()
 	defer r.reconcilingLock.Unlock()
+	r.destroyFunc()
```

Comment:
> Calling destroyFunc() here will cause the later Delete call racing against the etcd client destruction and cause a potential error.

It seems multi delete calls does not cause problem because the destroyFunc routine is protected by ```once```.
Ofcource I undestand we need to reduce duplicate destroy calls.
But now we need to fix a memory leak as workaroud.

https://github.com/kubernetes/kubernetes/blob/04b7977ef63247aa8e6b85fc5d12d3724ab8975c/staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go#L243-L252